diff --git a/configure.ac b/configure.ac
index a38e3fe..e02f199 100644
--- a/configure.ac
+++ b/configure.ac
@@ -60,15 +60,14 @@ case $host in
     platform_defs="-DXP_UNIX"
     platform="hpux"
     ;;
-  sparc-sun-solaris*)
+  *-*-solaris*)
     platform_defs="-DXP_UNIX"
     platform="solaris"
-dnl Cstd and Crun are required to link any C++ related code (ICU)
-    LIBCSTD=-lCstd
+dnl Cstd and Crun are not required
+    LIBCSTD=
     AC_SUBST([LIBCSTD], [$LIBCSTD])
-    LIBCRUN=-lCrun
+    LIBCRUN=
     AC_SUBST([LIBCRUN], [$LIBCRUN])
-    CXXLINK_REQUIRED=1
     ;;
   *)
     platform_defs=""
