diff --git a/lib/jss.def b/lib/jss.def
index 2f371ac..7074951 100644
--- a/lib/jss.def
+++ b/lib/jss.def
@@ -303,7 +303,6 @@ Java_org_mozilla_jss_CryptoManager_OCSPCacheSettingsNative;
 Java_org_mozilla_jss_CryptoManager_setOCSPTimeoutNative;
 Java_org_mozilla_jss_CryptoManager_verifyCertificateNowNative;
 Java_org_mozilla_jss_CryptoManager_verifyCertificateNowNative2;
-Java_org_mozilla_jss_CryptoManager_verifyCertificateNowCUNative;
 Java_org_mozilla_jss_asn1_ASN1Util_getTagDescriptionByOid;
 Java_org_mozilla_jss_ssl_SocketBase_setSSLVersionRange;
 Java_org_mozilla_jss_ssl_SSLSocket_setSSLVersionRangeDefault;
@@ -312,15 +311,3 @@ Java_org_mozilla_jss_pkcs11_PK11SymKey_setNickNameNative;
 ;+    local:
 ;+       *;
 ;+};
-;+JSS_4.3.1 {     # JSS 4.3.1 release
-;+    global:
-Java_org_mozilla_jss_pkcs11_PK11KeyPairGenerator_generateECKeyPairWithOpFlags;
-Java_org_mozilla_jss_pkcs11_PK11KeyPairGenerator_generateRSAKeyPairWithOpFlags;
-Java_org_mozilla_jss_pkcs11_PK11KeyPairGenerator_generateDSAKeyPairWithOpFlags;
-Java_org_mozilla_jss_CryptoManager_OCSPCacheSettingsNative;
-Java_org_mozilla_jss_CryptoManager_setOCSPTimeoutNative;
-Java_org_mozilla_jss_pkcs11_PK11SymmetricKeyDeriver_nativeDeriveSymKey;
-Java_org_mozilla_jss_pkcs11_PK11SymKey_setNickNameNative;
-;+    local:
-;+       *;
-;+};
