diff --git a/src/utils.c b/src/utils.c
index fee42ec..aea47f8 100644
--- a/src/utils.c
+++ b/src/utils.c
@@ -117,6 +117,7 @@ void randomize_buffer(char *buffer,int buffer_size){
         }
 
 
+static int my_create_socket(struct addrinfo *ai, const char *bind_address);
 /* opens a connection to a remote host */
 int my_connect(const char *host, struct sockaddr_storage * hostaddr, u_short port,
 		int address_family, const char *bind_address){
