diff --git a/lib/base/dnsdmain.cpp b/lib/base/dnsdmain.cpp
index 0a3afd7..06bdcdb 100644
--- a/lib/base/dnsdmain.cpp
+++ b/lib/base/dnsdmain.cpp
@@ -40,7 +40,7 @@ extern int getdomainname(char *, size_t);
 #else 
 extern int getdomainname(char *, int);
 #endif /* Linux */
-#if defined(HPUX) || defined(LINUX) || defined(SOLARIS_GCC) || defined(__FreeBSD__)
+#if defined(HPUX) || defined(LINUX) || defined(SOLARIS) || defined(__FreeBSD__)
 extern int gethostname (char *name, size_t namelen);
 #else
 extern int gethostname (char *name, int namelen);
