diff --git a/Makefile.am b/Makefile.am
index 4863d41..7d65973 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1747,7 +1747,7 @@ libschemareload_plugin_la_LDFLAGS = -avoid-version
 libstatechange_plugin_la_SOURCES = ldap/servers/plugins/statechange/statechange.c
 
 libstatechange_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(DSPLUGIN_CPPFLAGS)
-libstatechange_plugin_la_LIBADD = libslapd.la
+libstatechange_plugin_la_LIBADD = libslapd.la $(NSPR_LINK)
 libstatechange_plugin_la_DEPENDENCIES = libslapd.la
 libstatechange_plugin_la_LDFLAGS = -avoid-version
 
