diff --git a/src/libsds/test/test_sds_csiphash.c b/src/libsds/test/test_sds_csiphash.c
index cdb6b7f..75e0b1f 100644
--- a/src/libsds/test/test_sds_csiphash.c
+++ b/src/libsds/test/test_sds_csiphash.c
@@ -18,6 +18,20 @@
 #include <sys/endian.h>
 #elif defined(HAVE_ENDIAN_H)
 #include <endian.h>
+#elif defined(SOLARIS)
+#   include <sys/byteorder.h>
+#   define htobe16(x) BE_16(x)
+#   define htole16(x) LE_16(x)
+#   define be16toh(x) BE_IN16(x)
+#   define le16toh(x) LE_IN16(x)
+#   define htobe32(x) BE_32(x)
+#   define htole32(x) LE_32(x)
+#   define be32toh(x) BE_IN32(x)
+#   define le32toh(x) LE_IN32(x)
+#   define htobe64(x) BE_64(x)
+#   define htole64(x) LE_64(x)
+#   define be64toh(x) BE_IN64(x)
+#   define le64toh(x) LE_64(x)
 #else
 #error platform header for endian detection not found.
 #endif
