diff --git a/setup.py b/setup.py
index ec3a4be..0651cce 100644
--- a/setup.py
+++ b/setup.py
@@ -155,6 +155,7 @@ setup(
     'ldap.controls.deref',
     'ldap.controls.libldap',
     'ldap.controls.openldap',
+    'ldap.controls.pagedresults',
     'ldap.controls.ppolicy',
     'ldap.controls.psearch',
     'ldap.controls.pwdpolicy',
@@ -162,6 +163,7 @@ setup(
     'ldap.controls.sessiontrack',
     'ldap.controls.simple',
     'ldap.controls.sss',
+    'ldap.controls.vlv',
     'ldap.cidict',
     'ldap.dn',
     'ldap.extop',
